### ISSUE TITLE
[BUG] Random seed does not take effect during finetuning

### ISSUE THREAD

**Issue by zjgemi** [deepmodeling/deepmd-kit#3769]
### Bug summary

The parameter `seed` in `training` part does not take effect during finetuning. The results do not change after modifying the seed.

### DeePMD-kit Version

2024Q1

### Backend and its version

pytorch

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

dp --pt train --finetune pretrained_model.pt --model-branch H2O_H2O-PD input.json

### Steps to Reproduce

Modify the seed and rerun the command.
